src/tools/rpc.rs

```diff
@@ -73,6 +73,42 @@ pub async fn get_transaction_info(
         .await?)
 }
 
+pub async fn get_ledger_info() -> anyhow::Result<RpcResponse<LedgerInfoResponse>> {
+    let response = build_json_request(&build_ledger_info_request())
+        .send()
+        .await?;
+    Ok(response
+        .error_for_status()?
+        .json::<RpcResponse<LedgerInfoResponse>>()
+        .await?)
+}
+
+fn build_ledger_info_request() -> RpcRequest<Vec<LedgerInfoRequest>> {
```

Comment:
this helper doesn't seem practical either, and the `builder` part in the name is misleading (I was expecting the builder pattern)

Reply:
Removed